torchmultimodal/modules/encoders/mdetr_text_encoder.py

```diff
@@ -0,0 +1,284 @@
+        return states
+
+    def _forward_return_last_layer(
+        self,
+        embeddings: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+    ) -> torch.Tensor:
+        encoded = embeddings
+        for layer in self.layers.layers:
+            encoded = layer(encoded, src_key_padding_mask=attention_mask)
+        if self.normalize_before:
+            encoded = self.embedding_layer_norm(encoded)
+        return encoded
+
+    def forward(
+        self,
+        embeddings: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+    ) -> Union[torch.Tensor, List[torch.Tensor]]:
+        out: Union[torch.Tensor, List[torch.Tensor]]
+        if self.return_all_layers:
+            out = self._forward_return_all_layers(embeddings, attention_mask)
+        else:
+            out = self._forward_return_last_layer(embeddings, attention_mask)
+
+        return out
+
+
+class MDETRTextEncoder(nn.Module):
+    """
+    Text encoder for MDETR. Combines an embedding module with a transformer encoder.
+
+    Args:   embeddings (nn.Module): Embedding module (input IDs -> embeddings).
+            encoder (nn.Module): Transformer encoder module
+                (embeddings -> encoder outputs).
+
+    Inputs: input_ids (Tensor): Tensor of input IDs to encode.
+            token_type_ids (Optional[Tensor]): Optional tensor of token type IDs to use
+                in token type embedding. Default: None
+            attention_mask (Optional[Tensor]): Attention mask for batch. Should equal 1
+                on masked tokens on 0 on non-masked tokens. Default: None (no masking)
+            position_ids (Optional[Tensor]): Optional tensor of position IDs to use in
+                embeddings. Default: None
+            inputs_embeds (Optional[Tensor]): Embeddings of input IDs (useful if
+                already computed elsewhere). Default: None
+    """ ""
+
+    def __init__(self, embeddings: nn.Module, encoder: nn.Module):
+        super().__init__()
+        self.embeddings = embeddings
+        self.encoder = encoder
+
+    def forward(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        token_type_ids: Optional[torch.Tensor] = None,
+        position_ids: Optional[torch.Tensor] = None,
+    ):
+
+        embedding_output = self.embeddings(
+            input_ids=input_ids,
+            position_ids=position_ids,
+            token_type_ids=token_type_ids,
+        )
+        out = self.encoder(
+            embedding_output,
+            attention_mask=attention_mask,
+        )
+
+        return out
+
+
+def mdetr_roberta_text_encoder():
+    embeddings = MDETRTextEmbeddings(
+        hidden_size=768,
+        vocab_size=50265,
+        pad_token_id=1,
+        type_vocab_size=1,
+        max_position_embeddings=514,
+        layer_norm_eps=1e-05,
+        hidden_dropout_prob=0.1,
+    )
+
+    wrapped_args = {
```

Comment:
why do u need this logic?cant u just write them yourself and expose them as function args

Reply:
Not necessary, mainly just to make the dependence on TorchText RoBERTa more explicit. But can also remove these